src/main/java/duke/commands/AddDeadlineCommand.java

```diff
@@ -0,0 +1,65 @@
+package duke.commands;
+
+import java.io.IOException;
+
+import duke.main.DukeException;
+import duke.main.Parser;
+import duke.main.Storage;
+import duke.main.TaskList;
+import duke.main.Ui;
+import duke.tasks.Deadline;
+
+
+/**
+ * AddDeadlineCommand is a Command.
+ * This Command is used to add Deadlines to the TaskList.
+ */
+public class AddDeadlineCommand extends Command<String> {
+    private final Storage storage;
+
+    /**
+     * Constructor for AddDeadlineCommand.
+     * When this class is instantiated, it automatically runs runCommand().
+     *
+     * @param toDoList the user's List of Tasks
+     * @param cmd      the user input to Burp
+     * @param storage  the textfile used to store history of TaskList
+     * @throws DukeException if no Task description is given
+     */
+    public AddDeadlineCommand(TaskList toDoList, String cmd, Storage storage) throws DukeException {
+        this.storage = storage;
+        this.runCommand(toDoList, cmd);
```

Comment:
Calling a method works for this, but not conventional to call methods inside constructors as the method calls can not be seen in code calling the constructor and might get confusing. I suggest calling the constructor and then calling runCommand in the code manually after the constructor.